source/vistas/core/graphics/overlay.py

```diff
@@ -72,39 +72,37 @@ def generate_mesh(self):
         num_indices = 6 * len(self._buttons)
         num_vertices = 4 * len(self._buttons)
 
-        if self.mesh is None:
-            self.mesh = Mesh(num_indices, num_vertices, has_texture_coords=True, mode=Mesh.TRIANGLES)
-            self.mesh.shader = ShaderProgram()
-            self.mesh.shader.attach_shader(
-                os.path.join(get_resources_directory(), 'shaders', 'overlay_vert.glsl'), GL_VERTEX_SHADER
-            )
-            self.mesh.shader.attach_shader(
-                os.path.join(get_resources_directory(), 'shaders', 'overlay_frag.glsl'), GL_FRAGMENT_SHADER
-            )
+        if self.shader is None:
+            self.shader = ShaderProgram()
+            self.shader.attach_shader(get_builtin_shader('overlay_vert.glsl'), GL_VERTEX_SHADER)
+            self.shader.attach_shader(get_builtin_shader('overlay_frag.glsl'), GL_FRAGMENT_SHADER)
+            self.shader.link_program()
+
+        if self.geometry is None:
+            self.geometry = Geometry(num_indices, num_vertices, has_texture_coords=True, mode=Geometry.TRIANGLES)
 
-        # Create new mesh if buttons are added or removed; otherwise can use existing buffers
-        if self.mesh.num_indices != num_indices or self.mesh.num_vertices != num_vertices:
-            self.mesh = None
+        # Create new geometry if buttons are added or removed; otherwise can use existing buffers
+        if self.geometry.num_indices != num_indices or self.geometry.num_vertices != num_vertices:
+            self.geometry = None
             return self.generate_mesh()
 
-        self.mesh.vertices = numpy.array([
+        self.geometry.vertices = numpy.array([
             [
                 x.position[0], x.position[1], 0,
-                x.position[0]+x.size[0], x.position[1], 0,
-                x.position[0], x.position[1]+x.size[1], 0,
-                x.position[0]+x.size[0], x.position[1]+x.size[1], 0
+                x.position[0] + x.size[0], x.position[1], 0,
```

Comment:
Hmm I suppose so. This was following the PEP convention for spacing operators, but if we don't mind bending that rule I'm fine with undoing this.

Reply:
The previous formatting is [perfectly allowable in PEP-8](https://www.python.org/dev/peps/pep-0008/#other-recommendations).

> If operators with different priorities are used, consider adding whitespace around the operators with the lowest priority(ies). Use your own judgment; however, never use more than one space, and always have the same amount of whitespace on both sides of a binary operator.

```python
i = i + 1
submitted += 1
x = x*2 - 1
hypot2 = x*x + y*y
c = (a+b) * (a-b)
```

Reply:
Granted this case is a little different, since they're separate arguments and not different priority operators. Still, PEP-8 gives you the option of using space around operators, or not, as long as you use the same amount of space on both sides.

Reply:
Sounds good, I'll undo the formatting change.